### PR TITLE
[mesheryctl] Enhance `mesh remove` command

### LINKED ISSUE #3873: [mesheryctl] enhance `mesheryctl mesh remove`
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
`mesheryctl mesh remove` doesn't use the event streamer utility and thus doesn't wait for remove operation event to complete.

#### Desired Behavior
The command should watch for events and wait till receiving an event signalling service mesh was removed successfully. If it is not removed, then the command should throw an error. 

#### Implementation
The implementation should be similar to the `mesheryctl mesh deploy` command. For reference, see
https://github.com/meshery/meshery/blob/c8dd33ae4f8756b0f69f06edb27e6cda5b904498/mesheryctl/internal/cli/root/mesh/deploy.go#L122-L173

#### Acceptance Tests
Exit with proper exit codes when operation fails.

#### Mockups

---

#### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**
This PR is WIP

This PR fixes #3873

**Notes for Reviewers**
Based on `mesh deploy` functionality. Maintainers/Members I need your review for this

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/mesh/remove.go [hunk 1]
```diff
@@ -70,4 +81,52 @@ func init() {
 	removeCmd.Flags().StringVarP(&namespace, "namespace", "n", "default", "Kubernetes namespace to be used for deploying the validation tests and sample workload")
 	removeCmd.Flags().StringVarP(&tokenPath, "tokenPath", "t", "", "Path to token for authenticating to Meshery API")
 	_ = removeCmd.MarkFlagRequired("tokenPath")
+	removeCmd.Flags().BoolVarP(&watch, "watch", "w", false, "Watch for events and verify operation (in beta testing)")
+}
+
+func waitForRemoveResponse(mctlCfg *config.MesheryCtlConfig, query string) (string, error) {
+	path := mctlCfg.GetBaseMesheryURL() + "/api/events?client=cli_remove"
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/mesh/remove.go:88]: @leecalcote @DelusionalOptimist I checked with watch flag to check if this function works... But it threw error (of course mesh is removed but GET request threw error while testing). Is the API call URL correct?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: @mgfeller with respect to the upcoming focus on Messaging Framework, it'd be good to familiarize with the /api/events. Gosh, it's been so long that I'm not sure what the expected behavior here is.

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: I referred the REST API docs too to get clarified https://docs.meshery.io/reference/rest-apis ... But no luck, maybe I used a wrong URL by mistake 😬

Comment (leecalcote) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: @piyushsingariya might assist here.

Comment (piyushsingariya) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: This URL isn't covered in the docs, will check if the URL is correct and get back here soon.

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: @piyushsingariya is the URL thing solved? // Just pinging for update

Comment (piyushsingariya) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: URL is correct. You must be getting any errors if it's not 200. Could you paste that here?

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: I'll check again and get back to you...

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: Not error codes... but GET request error...
image

Comment (alphaX86) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: And it pauses after that, ending in infinite loop....

Comment (piyushsingariya) [reply mesheryctl/internal/cli/root/mesh/remove.go:88]: @utkarsh-pro Help!!

**Comment by alphaX86** [comment]
> @alphaX86 Were you able to look into this?

That GET request is driving me crazy... IDK where it went wrong yet, plus this other works.. :')

**Comment by navendu-pottekkat** [comment]
@alphaX86 What are the other pending works here? Do you have any specific questions I can help with?

**Comment by alphaX86** [comment]
@navendu-pottekkat or @piyushsingariya could you review and check if the `watch` flag works as intended?

**Comment by warunicorn19** [comment]
@navendu-pottekkat @piyushsingariya did you guys get a chance to review this?

**Comment by alphaX86** [comment]
Hey @warunicorn19 if possible could you try my PR and check if `--watch` flag works with no barriers? I tested it but I'm still in dilemma if I made the right changes. 80% is fixed but I need to cross-check it

**Comment by warunicorn19** [comment]
Sure @alphaX86, I'll take a look.

**Comment by alphaX86** [comment]
@navendu-pottekkat what's the status? You said you'd look on this! 🙃

**Comment by navendu-pottekkat** [comment]
@alphaX86 I couldn't reach to a conclusion on what the issue might be here. All the commands that use the events streamer is stuck at verifying operation. Could you check with a locally running Meshery server. I will do the same. Let's check if the logs gives any info.

On a side note, could you add the default token path here? Check the `perf` command and make sure that the token flag is the same throughout and it defaults to `.meshery/auth.json`.

**Comment by alphaX86** [comment]
I understand, let's do it and check if it complies to the requests or not

**Comment by navendu-pottekkat** [comment]
@alphaX86  It was successful when running Meshery and the adapter locally. Were there any changes in recently related to this that hasn't made to the release?
 
Could you try to debug this?

Screenshot 2021-11-13 at 9 43 03 PM

**Comment by navendu-pottekkat** [comment]
So, the PR should be good after you update the token flag as I mentioned before and debug this issue.

**Comment by alphaX86** [comment]
> @alphaX86 It was successful when running Meshery and the adapter locally. Were there any changes in recently related to this that hasn't made to the release?
> 
> Could you try to debug this?

Oh ok 👀 so when Meshery is made to run locally, it worked?! OK I'll debug today on the ones you've mentioned

**Comment by alphaX86** [comment]
Closing this as this is gone way beyond outdated; I'll open a new one with fresh changes when free...

**Comment by leecalcote** [comment]
Please do, @alphaX86 👍

**Comment by leecalcote** [comment]
Ping when you have done so.

**Comment by alphaX86** [comment]
Rog' I may recomplete this on Sat/Sunday as I'd be free on those days for this week :')

**Pull request closed without merging**